Add support for transforming boolean tests

hannes opened the pull request:
Allow checks for `x IS [NOT] TRUE/FALSE/NULL/UNKNOWN`

hannes committed:
add support for transforming boolean tests, allow checks for IS [NOT] TRUE/FALSE/NULL/UNKNOWN

Tishj commented:
Can you also add this patch?

```patch
From 397454812bf8aca3ec64323ffbb0b538211babcc Mon Sep 17 00:00:00 2001
From: Tishj <t_b@live.nl>
Date: Thu, 30 Mar 2023 18:55:01 +0200
Subject: [PATCH] fix #6916 - use NodetypeToString

---
 src/parser/transform/expression/transform_expression.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/parser/transform/expression/transform_expression.cpp b/src/parser/transform/expression/transform_expression.cpp
index 5c992d28ce..0e3a146d7b 100644
--- a/src/parser/transform/expression/transform_expression.cpp
+++ b/src/parser/transform/expression/transform_expression.cpp
@@ -80,7 +80,7 @@ unique_ptr<ParsedExpression> Transformer::TransformExpression(duckdb_libpgquery:
 	case duckdb_libpgquery::T_PGAStar:
 		return TransformStarExpression(node);
 	default:
-		throw NotImplementedException("Expr of type %d not implemented\n", (int)node->type);
+		throw NotImplementedException("Expr of type '%s' not implemented\n", NodetypeToString(node->type));
 	}
 }
 
-- 
2.37.1 (Apple Git-137.1)
```
Should make the error message a little less cryptic for future issues

hannes committed:
attempt at a more descriptive error message for unknown expression types

hannes commented:
Yup, added a slightly derived version. Since the `NotImplementedException` already adds "Not Implemented", the text can be shortened and to be sure I've also added the number as a fall-back when the stringification fails.

Mytherin commented:
Thanks! LGTM
